src/fsm/parser.act

```diff
@@ -90,6 +99,58 @@
 		exit(EXIT_FAILURE);
 	}
 
+	/* FNV1a 32-bit hash function, in the public domain. For details,
+	 * see: http://www.isthe.com/chongo/tech/comp/fnv/index.html . */
+	#define FNV1a_32_OFFSET_BASIS 	2166136261UL
+	#define FNV1a_32_PRIME		16777619UL
+
+	static unsigned
+	hash_of_id(const char *id)
+	{
+		unsigned h = FNV1a_32_OFFSET_BASIS;
+		while (*id) {
+			h ^= *id;
+			h *= FNV1a_32_PRIME;
+			id++;
+		}
+		return h;
+	}
+
+	static void
+	grow_state_hash_table(struct act_stateset *htab)
+	{
+		const size_t ncount = 2 * htab->bucket_count;
+		struct act_statelist **nbuckets = xcalloc(ncount, sizeof(*nbuckets));
+		size_t max_chain_length = 0;
+		size_t old_i;
+		unsigned mask = ncount - 1;
+		if (nbuckets == NULL) {
```

Comment:
This can never happen now (because xcalloc will exit)

Reply:
Fixed, amended.